### PR TITLE
klaus: patch for python 3.12

### PR DESCRIPTION
**Pull request by fin444**
## Description of changes

See https://github.com/jonashaag/klaus/pull/334
Not sure how long that will take to merge and then make it to a new version, so putting this in to fix for now

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/klaus/default.nix [hunk 1]
```diff
@@ -52,6 +52,8 @@ buildPythonPackage rec {
   # Needs to set up some git repos
   doCheck = false;
 
+  patches = [ ./distutils.patch ];
```
Comment (kirillrdy) [inline pkgs/development/python-modules/klaus/default.nix:55]: any reasons not to use fetchpatch instead ?

Comment (fin444) [reply pkgs/development/python-modules/klaus/default.nix]: Wasn't aware it existed. I've updated to use fetchpatch

File: pkgs/development/python-modules/klaus/default.nix [hunk 2]
```diff
@@ -52,6 +53,15 @@ buildPythonPackage rec {
   # Needs to set up some git repos
   doCheck = false;
 
+  # TODO: remove when this is upstreamed
+  patches = [
```
Comment (kirillrdy) [inline pkgs/development/python-modules/klaus/default.nix:57]: `patches` is usually located after `src`

File: pkgs/development/python-modules/klaus/default.nix [hunk 3]
```diff
@@ -52,6 +53,15 @@ buildPythonPackage rec {
...
+    (fetchpatch {
+      name = "distutils.patch";
+      url = "https://github.com/jonashaag/klaus/commit/d50d2aab97fd86c11f3b5a4c1ecbcf1e085f395f.patch";
```
Comment (kirillrdy) [inline pkgs/development/python-modules/klaus/default.nix:60]: can you can a link to PR in a comment above ?

File: pkgs/development/python-modules/klaus/default.nix (hunk 2 above)
Comment (fin444) [reply pkgs/development/python-modules/klaus/default.nix]: I moved them to after `prepatch` so it's chronological-ish

File: pkgs/development/python-modules/klaus/default.nix (hunk 3 above)
Comment (fin444) [reply pkgs/development/python-modules/klaus/default.nix]: PR has been merged, now it's just next release. Comment updated.

**Review by kirillrdy — APPROVED** [review]
bonus points for next PR, klaus can be moved out of python-modules as it's only meant to be used as an application

**Comment by fin444** [comment]
Just gonna wait for the mass migration to by-name

**Comment by kirillrdy** [comment]
> Just gonna wait for the mass migration to by-name

mass migration is not going to touch python-modules

**Pull request merged**